### PR TITLE
fix(orbctl): add repokey parameter to configure command

### PR DESCRIPTION
**Pull request by stebenz**

### CODE REVIEW

**Review by eliobischof — APPROVED** [review]

File: pkg/git/client.go [hunk 1]
```diff
@@ -150,6 +150,7 @@ func (g *Client) writeCheck() (err error) {
 		if ref, err = g.repo.Tag(localWriteCheckTag); err != nil {
 			return err
 		}
+		createErr = nil
```
Comment (eliobischof) [inline pkg/git/client.go:153]: So we assume that an ErrTagExists error can only occur if we have write permissions?

File: pkg/git/client.go [hunk 2]
```diff
@@ -162,7 +163,7 @@ func (g *Client) writeCheck() (err error) {
 			config.RefSpec("+" + ref.Name() + ":" + ref.Name()),
 		},
 		Auth: g.auth,
-	}); pushErr != nil && pushErr == gogit.NoErrAlreadyUpToDate {
+	}); pushErr != nil && pushErr != gogit.NoErrAlreadyUpToDate {
 		return fmt.Errorf("write-check failed: %w", pushErr)
 	}
```
Comment (eliobischof) [inline pkg/git/client.go:169]:
```suggestion
	}); pushErr != nil && pushErr != gogit.NoErrAlreadyUpToDate {
		return fmt.Errorf("write-check failed: %w", pushErr)
	}
      // explicitly ignore other push errors
      pushErr = nil
```

File: pkg/orb/orb.go [hunk 3]
```diff
@@ -164,7 +179,8 @@ func Reconfigure(ctx context.Context, monitor mntr.Monitor, orbConfig *Orb, newR
 
 	// If the repokey already has read/write permissions, don't generate a new one.
 	// This ensures git providers other than github keep being supported
-	if err := configureGit(false); err != nil {
+	// Only if you're not trying to set a new key, as you don't want to generate a new key then
+	if err := configureGit(false); err != nil && newRepoKey == "" {
```
Comment (eliobischof) [inline pkg/orb/orb.go]: if newRepoKey != "", configureGit(true) should be called immediately and return the error if not nil

File: pkg/git/client.go (hunk 1 above)
Comment (stebenz) [reply pkg/git/client.go:153]: No, that we check write permission again with this function but the createErr is then irrelevant. Like if the second try succeeds you can ignore the error of the first one, but we only try again when the tag already exists.

File: pkg/git/client.go (hunk 2 above)
Comment (stebenz) [reply pkg/git/client.go:169]: Did you test it in the code? As pushErr is only declared inside this if-statement.

Comment (eliobischof) [reply pkg/git/client.go:169]: pushErr = nil shows the reader immediately that other pushErrs have no effect

Comment (eliobischof) [reply pkg/git/client.go:169]: I see, all errors are returned except gogit.NoErrAlreadyUpToDate. So it's okay for me.

**Review by eliobischof — APPROVED** [review]

File: pkg/orb/orb.go (hunk 3 above)
Comment (stebenz) [reply pkg/orb/orb.go]: Changed to only generate ssh-keys if it's github and no specific key is given as parameter.

**Pull request merged**